src/robot/Client.ts

```diff
@@ -325,14 +325,26 @@ export class RobotClient implements Robot {
         this.transportFactory = webRTCConn.transportFactory;
 
         webRTCConn.peerConnection.ontrack = (event) => {
-          events.emit('track', event);
-
           const { kind } = event.track;
 
           const eventStream = event.streams[0];
           if (!eventStream) {
+            events.emit('track', event);
             throw new Error('expected event stream to exist');
           }
+
+          // track id has +s to conform to RFC 4566 (https://www.rfc-editor.org/rfc/rfc4566)
+          // where names should not contain colons.
+          const resName = eventStream.id.replaceAll('+', ':');
```

Comment:
nit: let's put `getValidSDPTrackName` into utils and call it here too just so the logic is centralized?

Reply:
this does the reverse and I feel like these should actually be private functions - we don't want users to need to interact with these functions